docs/control-structures.rst

```diff
@@ -531,6 +531,12 @@ and will wrap without an error if used inside an unchecked block:
     It is not possible to disable the check for division by zero
     or modulo by zero using the ``unchecked`` block.
 
+.. note::
+   Bitwise operators do not perform overflow or underflow checks.
+   This is particularly visible when using bitwise shifts (``<<``, ``>>``, ``<<=``, ``>>=``) in
+   place of integer division and multiplication by a power of 2.
+   For example ``type(uint256).max << 3`` does not revert even though ``type(uint256).max * 8`` would.
```

Comment:
Did you check that the operation is actually performed in that type instead of in unlimited precision because of constants?

Reply:
It's performed in `uint256`. It only tries to use unlimited precision (and fails at compilation time) if I use a literal (i.e `(2**256 - 1)`).

Just in case I rechecked it in Remix (on 0.8.1):

```solidity
contract C {
    function foo() public {
        uint x = type(uint256).max * 8;    // Revert
        type(uint256).max * 8;             // Revert
        (2**256 - 1) * 8;                  // Compilation error: Invalid rational number
        uint y = 2**256 - 1; y * 8;        // Revert
    }

    function bar() public {
        uint x = type(uint256).max << 3;   // OK
        type(uint256).max << 3;            // OK
        (2**256 - 1) << 3;                 // Compilation error: Invalid rational number
        uint y = 2**256 - 1; y << 3;       // OK
    }
}
```

Reply:
ok, good! Yeah, I think it makes sense that `type(uint16).max` is of type `uint16`.